crates/primitives-traits/src/tx_type.rs

```diff
@@ -30,26 +30,18 @@ pub trait TxType:
     + alloy_rlp::Encodable
     + alloy_rlp::Decodable
 {
-}
+    /// Returns `true` if this is a legacy transaction.
+    fn is_legacy(&self) -> bool;
 
-impl<T> TxType for T where
-    T: Send
-        + Sync
-        + Unpin
-        + Clone
-        + Copy
-        + Default
-        + fmt::Debug
-        + fmt::Display
-        + PartialEq
-        + Eq
-        + PartialEq<u8>
-        + Into<u8>
-        + Into<U8>
-        + TryFrom<u8, Error: fmt::Debug>
-        + TryFrom<u64, Error: fmt::Debug>
-        + TryFrom<U64>
-        + alloy_rlp::Encodable
-        + alloy_rlp::Decodable
-{
+    /// Returns `true` if this is an eip-2930 transaction.
+    fn is_eip2930(&self) -> bool;
+
+    /// Returns `true` if this is an eip-1559 transaction.
+    fn is_eip1559(&self) -> bool;
+
+    /// Returns `true` if this is an eip-4844 transaction.
+    fn is_eip4844(&self) -> bool;
+
+    /// Returns `true` if this is an eip-7702 transaction.
+    fn is_eip7702(&self) -> bool;
```

Comment:
actually not sure if we call these on the txtype itself, we def call them on the tx. 
so I assume we could upstream these to alloy, probably as a standalone new trait

Reply:
we have match statements on `TxType`, which these would replace. Agree that they should be added to the tx trait too, along with the useful `as_eipX -> Option<&tx>` methods